sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -164,23 +233,20 @@ case class Invoke(
       """
     }
 
-    val setIsNull = if (propagateNull && arguments.nonEmpty) {
-      s"boolean ${ev.isNull} = ${obj.isNull} || ${argGen.map(_.isNull).mkString(" || ")};"
-    } else {
-      s"boolean ${ev.isNull} = ${obj.isNull};"
-    }
-
     // If the function can return null, we do an extra check to make sure our null bit is still set
     // correctly.
     val postNullCheck = if (ctx.defaultValue(dataType) == "null") {
       s"${ev.isNull} = ${ev.value} == null;"
     } else {
       ""
     }
+
     val code = s"""
       ${obj.code}
-      ${argGen.map(_.code).mkString("\n")}
-      $setIsNull
+      if (!${obj.isNull}) {
+        $argCode
+      }
+      boolean ${ev.isNull} = ${obj.isNull} || $resultIsNull;
```

Comment:
I modified the code as you mentioned.

But I'm sorry, I didn't understand what you want to do.

- You are assuming that the argument code is not so long, so use local variable without splitting evaluations? (could also make the argument variable as local variable)
  - I think the argument code would easily become large enough to exceed the 64kb limit.
- You want to modify only `resultIsNull` to local variable and pass it and return the value to the split methods?
  - We need to extend the method `ctx.splitExpressions()` to be able to pass and return value and handle it as #15480 is trying.
- You want to switch local variable or member variable by whether split or not?
  - We need to know whether split or not but we can't now. Modify `ctx.splitExpressions()` to return split or not?

Could you let me know if you have other thoughts?